src/thunderbots/software/proto/src/ssl_referee.proto

```diff
@@ -0,0 +1,193 @@
+        // If no yellow cards are issued, this array has no elements.
+        // Otherwise, times are ordered from smallest to largest.
+        repeated uint32 yellow_card_times = 4 [packed = true];
+        // The total number of yellow cards ever issued to the team.
+        required uint32 yellow_cards = 5;
+        // The number of timeouts this team can still call.
+        // If in a timeout right now, that timeout is excluded.
+        required uint32 timeouts = 6;
+        // The number of microseconds of timeout this team can use.
+        required uint32 timeout_time = 7;
+        // The pattern number of this team's goalie.
+        required uint32 goalie = 8;
+        // The total number of countable fouls that act towards yellow cards
+        optional uint32 foul_counter = 9;
+        // The number of consecutive ball placement failures of this team
+        optional uint32 ball_placement_failures = 10;
+        // Indicate if the team is able and allowed to place the ball
+        optional bool can_place_ball = 12;
+        // The maximum number of bots allowed on the field based on division and cards
+        optional uint32 max_allowed_bots = 13;
+    }
+
+    // Information about the two teams.
+    required TeamInfo yellow = 7;
+    required TeamInfo blue = 8;
+
+    // The coordinates of the Designated Position. These are measured in
+    // millimetres and correspond to SSL-Vision coordinates. These fields are
+    // always either both present (in the case of a ball placement command) or
+    // both absent (in the case of any other command).
+    message Point {
+        required float x = 1;
+        required float y = 2;
+    }
+    optional Point designated_position = 9;
+
+    // Information about the direction of play.
+    // True, if the blue team will have it's goal on the positive x-axis of the ssl-vision coordinate system.
+    // Obviously, the yellow team will play on the opposite half.
+    optional bool blue_team_on_positive_half = 10;
+
+    // The game event that caused the referee command.
+    // deprecated in favor of game_events.
+    optional Game_Event game_event = 11 [deprecated = true];
+
+    // The command that will be issued after the current stoppage and ball placement to continue the game.
+    optional Command next_command = 12;
+
+    // All game events that were detected since the last RUNNING state.
+    // Will be cleared as soon as the game is continued.
+    repeated GameEvent game_events = 13;
+
+    // All non-finished proposed game events that may be processed next.
+    repeated ProposedGameEvent proposed_game_events = 14;
+}
+
+message ProposedGameEvent {
+    // The UNIX timestamp when the game event proposal will time out, in microseconds.
+    required uint64 valid_until = 1;
+    // The identifier of the proposer.
+    required string proposer_id = 2;
+    // The proposed game event.
+    required GameEvent game_event = 3;
+}
```

Comment:
nit; github complaining about end-of-file

Reply:
fixed